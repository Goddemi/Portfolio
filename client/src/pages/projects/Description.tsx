import React from "react";
import styled from "styled-components";

import ProjectType from "./type/ProjectType";

const Description: React.FC<ProjectType> = ({
  summary,
  mainFunction,
  skills,
  git,
  result,
}) => {
  return (
    <Container>
      <Summary>{summary}</Summary>
      <Box>
        <span>주오 기능 : </span>
        <span>{mainFunction}</span>
      </Box>
      <Box>
        <span>사용된 기술 : </span>
        <span>{skills}</span>
      </Box>

      <Box>
        <span>Git : </span>
        <span>{git}</span>
      </Box>

      <Box>
        <span>결과물</span>
        <span>{result}</span>
      </Box>
    </Container>
  );
};

export default Description;

const Container = styled.div`
  width: 400px;
  margin-left: 30px;
`;

const Summary = styled.div`
  background-color: aqua;
  word-wrap: break-word;
  line-height: 20px;
`;

const Box = styled.div`
  margin: 10px 0;
`;
