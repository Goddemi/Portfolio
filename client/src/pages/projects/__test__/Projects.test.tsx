import { render, screen } from "@testing-library/react";
import { projectGenerator } from "../../../test/data-gererators";
import Description from "../Description";

describe("<Description />", () => {
  test("render project informantion in Description", () => {
    const { summary, main_function, skills, git, result } = projectGenerator();
    render(
      <Description
        summary={summary}
        main_function={main_function}
        skills={skills}
        git={git}
        result={result}
      />
    );

    const summaryText = screen.getByText(summary);
    expect(summaryText).toBeInTheDocument();
  });
});
