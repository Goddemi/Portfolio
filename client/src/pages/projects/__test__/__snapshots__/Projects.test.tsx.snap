// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Description /> project page renders correctly 1`] = `
<div
  className="sc-bcXHqe gcGRGz"
>
  <div
    className="sc-gswNZR bHGOZB"
  >
    xup=+Tkz/;
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      주요 기능 : 
    </span>
    <div>
      - 
      -hz@ftztkk
    </div>
    <div>
      - 
      99634
    </div>
    <div>
      - 
      24798
    </div>
    <div>
      - 
      ^\`gp7qI@*0
    </div>
    <div>
      - 
      "a^imbf%NG
    </div>
    <div>
      - 
      Cv6k7F_ipO
    </div>
    <div>
      - 
      65644
    </div>
    <div>
      - 
      7Xez;FP6{k
    </div>
    <div>
      - 
      6M&gt;_HP]YDT
    </div>
    <div>
      - 
      81693
    </div>
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      사용된 기술 : 
    </span>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      86468
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      48058
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      CPtaT=OG:;
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      85352
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      6)qXD("Pz"
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      MyWd?16*}p
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      fH(6hP;2T\`
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      0eqwMT5.x:
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      7#BB%jDBy\\
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      sP:N.P%kS;
    </div>
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      Git : 
    </span>
    <a
      href="dense-nobody.name"
    >
      GitLink
    </a>
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      Result : 
    </span>
    <a
      href="zesty-calendar.name"
    >
      SiteLink
    </a>
  </div>
</div>
`;

exports[`Project Page <Description /> project page renders correctly 1`] = `
<div
  className="sc-bcXHqe gcGRGz"
>
  <div
    className="sc-gswNZR bHGOZB"
  >
    d\\iIl&!rs#
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      주요 기능 : 
    </span>
    <div>
      - 
      98390
    </div>
    <div>
      - 
      S#Ln65ZZv4
    </div>
    <div>
      - 
      62865
    </div>
    <div>
      - 
      41975
    </div>
    <div>
      - 
      12505
    </div>
    <div>
      - 
      ^iQ1bd:BX:
    </div>
    <div>
      - 
      '89,M-2pbd
    </div>
    <div>
      - 
      47023
    </div>
    <div>
      - 
      48324
    </div>
    <div>
      - 
      \\)I)Q|Oz:t
    </div>
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      사용된 기술 : 
    </span>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      |;e'KQG^:n
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      89326
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      RP@5_&lt;k9'0
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      68284
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      82180
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      8027
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      k9#'&lt;()9Hx
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      87654
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      *XNhVCLJvK
    </div>
    <div
      className="sc-eDvSVe feYcqJ"
    >
      32980
    </div>
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      Git : 
    </span>
    <a
      href="taut-pitcher.info"
    >
      GitLink
    </a>
  </div>
  <div
    className="sc-dkrFOg dyrSQj"
  >
    <span
      className="sc-hLBbgP fHywhL"
    >
      Result : 
    </span>
    <a
      href="twin-bargain.biz"
    >
      SiteLink
    </a>
  </div>
</div>
`;
